test/test-script.py

```diff
@@ -58,31 +66,67 @@ def assert_running(unit):
+assert_matches(f"ip netns exec nb-spark-wallet curl -s {spark_auth}@169.254.1.17:9737", "Spark")
 
 assert_running("lightning-charge")
 charge_auth = re.search("API_TOKEN=(.*)", succeed("cat /secrets/lightning-charge-env"))[1]
-machine.wait_for_open_port(9112)
-assert_matches(f"curl -s api-token:{charge_auth}@localhost:9112/info | jq", '"id"')
+machine.wait_until_succeeds("ip netns exec nb-nanopos nc -z 169.254.1.18 9112")
+assert_matches(
+    f"ip netns exec nb-nanopos curl -s api-token:{charge_auth}@169.254.1.18:9112/info | jq", '"id"',
+)
 
 assert_running("nanopos")
-machine.wait_for_open_port(9116)
-assert_matches("curl localhost:9116", "tshirt")
+machine.wait_until_succeeds("ip netns exec nb-lightning-charge nc -z 169.254.1.19 9116")
+assert_matches("ip netns exec nb-lightning-charge curl 169.254.1.19:9116", "tshirt")
 
 assert_running("onion-chef")
 
 # FIXME: use 'wait_for_unit' because 'create-web-index' always fails during startup due
 # to incomplete unit dependencies.
 # 'create-web-index' implicitly tests 'nodeinfo'.
 machine.wait_for_unit("create-web-index")
-machine.wait_for_open_port(80)
-assert_matches("curl localhost", "nix-bitcoin")
-assert_matches("curl -L localhost/store", "tshirt")
+machine.wait_until_succeeds("ip netns exec nb-nginx nc -z localhost 80")
+assert_matches("ip netns exec nb-nginx curl localhost", "nix-bitcoin")
+assert_matches("ip netns exec nb-nginx curl -L localhost/store", "tshirt")
 
 machine.wait_until_succeeds(log_has_string("bitcoind-import-banlist", "Importing node banlist"))
 assert_no_failure("bitcoind-import-banlist")
 
+### Security tests
+
+# Positive ping tests
+machine.succeed(
+    "ip netns exec nb-bitcoind ping -c 1 -w 1 169.254.1.12 && \
+    ip netns exec nb-bitcoind ping -c 1 -w 1 169.254.1.13 && \
+    ip netns exec nb-bitcoind ping -c 1 -w 1 169.254.1.15 && \
+    ip netns exec nb-nanopos ping -c 1 -w 1 169.254.1.18 && \
+    ip netns exec nb-nanopos ping -c 1 -w 1 169.254.1.19"
+)
+
+# Negative ping tests
+machine.fail(
+    "ip netns exec nb-bitcoind ping -c 1 -w 1 169.254.1.17 || \
+    ip netns exec nb-bitcoind ping -c 1 -w 1 169.254.1.18 || \
+    ip netns exec nb-bitcoind ping -c 1 -w 1 169.254.1.19 || \
+    ip netns exec nb-bitcoind ping -c 1 -w 1 169.254.1.20 || \
+    ip netns exec nb-bitcoind ping -c 1 -w 1 169.254.1.21 || \
+    ip netns exec nb-nanopos ping -c 1 -w 1 169.254.1.12 || \
+    ip netns exec nb-nanopos ping -c 1 -w 1 169.254.1.13 || \
+    ip netns exec nb-nanopos ping -c 1 -w 1 169.254.1.14 || \
+    ip netns exec nb-nanopos ping -c 1 -w 1 169.254.1.15 || \
+    ip netns exec nb-nanopos ping -c 1 -w 1 169.254.1.16 || \
+    ip netns exec nb-nanopos ping -c 1 -w 1 169.254.1.17 || \
+    ip netns exec nb-nanopos ping -c 1 -w 1 169.254.1.20"
+)
+
+# netns-exec tests
+machine.fail("netns-exec nb-electrs ip a")
+assert_matches_exactly(
+    "su operator -c 'netns-exec nb-bitcoind capsh --print | grep Current | tr -d \"\\n\"'",
+    "Current: =",
```

Comment:
Can't you match against "Current: =\n" instead of using `tr`?

Reply:
Fixed in a3aaff192b1988064f22ad77a25514c2e535ee1f